pkg/controller/llmisvc/router.go

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright 2025 The KServe Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package llmisvc
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/utils/ptr"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/kmeta"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gatewayapi "sigs.k8s.io/gateway-api/apis/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/kserve/kserve/pkg/apis/serving/v1alpha1"
+)
+
+func (r *LLMInferenceServiceReconciler) reconcileRouter(ctx context.Context, llmSvc *v1alpha1.LLMInferenceService) error {
+	logger := log.FromContext(ctx)
+
+	logger.Info("Reconciling Router")
+
+	if err := r.reconcileScheduler(ctx, llmSvc); err != nil {
+		llmSvc.MarkRouterNotReady("SchedulerReconcileError", "Failed to reconcile scheduler: %v", err.Error())
+		return fmt.Errorf("failed to reconcile workload: %w", err)
+	}
+
+	if err := r.reconcileGateway(ctx, llmSvc); err != nil {
+		llmSvc.MarkRouterNotReady("GatewayReconcileError", "Failed to reconcile gateway: %v", err.Error())
+		return err
+	}
+
+	// TODO: reconcile networkingv1.Ingress
+
+	if err := r.reconcileHTTPRoutes(ctx, llmSvc); err != nil {
+		llmSvc.MarkRouterNotReady("HTTPRouteReconcileError", "Failed to reconcile HTTPRoute: %v", err.Error())
+		return err
```

Comment:
do we want to augment the error message here?

Reply:
Wrapped them in 031152e, thx